### PR TITLE
Functional tests - Product settings enable/disable display unavailable product attributes

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Enable/Disable display unavailable product attribute
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/03_productSettings/02_productPage/03_displayUnavailableProductAttributes.js" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17891)

### CODE REVIEW

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/03_displayUnavailableProductAttributes.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        test.args.enable,
+      );
+      await expect(result).to.contains(this.pageObjects.productSettingsPage.successfulUpdateMessage);
+    });
+
+    it('should check the unavailable product attributes in FO product page', async function () {
+      await testContext.addContextItem(
+        this,
+        'testIdentifier',
+        `checkUnavailableAttribute_${test.args.action}`,
```
Comment (PierreRambaud) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/03_displayUnavailableProductAttributes.js]:
```suggestion
        `checkUnavailableAttribute${test.args.action[0].toUpperCase()}${test.args.action.slice(1)}`,
```
Or create a function for doing a uppercase on the first character. But don't mix snake case and camel case

**Review by NeOMakinG — DISMISSED** [review]

File: tests/puppeteer/pages/commonPage.js [hunk 2]
```diff
@@ -336,4 +336,13 @@ module.exports = class CommonPage {
...
     await this.page.mouse.up();
   }
+
+  /**
+   * Uppercase the first character of the word
+   * @param word
+   * @returns {string}
+   */
+  uppercaseFirstCharacter(word) {
+    return word[0].toUpperCase() + word.slice(1);
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/commonPage.js]:
```suggestion
    return `${word[0].toUpperCase()}${word.slice(1)}`;
```

File: tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/03_displayUnavailableProductAttributes.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+  });
+
+  it('should delete product', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'deleteProduct', baseContext);
+    const deleteTextResult = await this.pageObjects.productsPage.deleteProduct(productData);
+    await expect(deleteTextResult).to.equal(this.pageObjects.productsPage.productDeletedSuccessfulMessage);
+  });
+
+  it('should reset all filters', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'resetFilters2', baseContext);
```
Comment (Progi1984) [inline tests/puppeteer/campaigns/functional/BO/13_shopParameters/03_productSettings/02_productPage/03_displayUnavailableProductAttributes.js]:
```suggestion
    await testContext.addContextItem(this, 'testIdentifier', 'resetAllFilters', baseContext);
```

File: tests/puppeteer/pages/BO/shopParameters/productSettings/index.js [hunk 4]
```diff
@@ -106,9 +107,25 @@ module.exports = class productSettings extends BOBasePage {
...
     await this.clickAndWaitForNavigation(this.saveProductPageFormButton);
     return this.getTextContent(this.alertSuccessBlock);
   }
+
+  /**
+   * Set display unavailable product attributes
+   * @param toEnable
+   * @returns {Promise<string>}
+   */
+  async setDisplayUnavailableProductAttributes(toEnable = true) {
```
Comment (Progi1984) [inline tests/puppeteer/pages/BO/shopParameters/productSettings/index.js]:
```suggestion
  async setDisplayUnavailableProductAttributesStatus(toEnable = true) {
```

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]
lgtm

**Review by Progi1984 — APPROVED** [review]

**Pull request merged**